testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -2360,4 +2362,197 @@ mod test {
+
+        _m.assert();
+    }
+
+    #[test]
+    fn test_event_dispatcher_tx_filtering_wrong() {
+        let dir = tempdir().unwrap();
+        let working_dir = dir.path().to_path_buf();
+
+        // Create a mock server
+        let mut server = mockito::Server::new();
+        let _m = server
+            .mock("POST", "/new_mempool_tx")
+            .with_status(200)
+            .expect_at_most(0)
+            .create();
+
+        let mut event_dispatcher = EventDispatcher::new();
+        let config = EventObserverConfig {
+            endpoint: server.url().strip_prefix("http://").unwrap().to_string(),
+            events_keys: vec![EventKeyType::BlockProposal],
+            timeout_ms: 1000,
+        };
+        event_dispatcher.register_observer(&config, working_dir);
+
+        let sig = SinglesigSpendingCondition {
+            hash_mode: SinglesigHashMode::P2WPKH,
+            signer: Hash160([0u8; 20]),
+            nonce: 0,
+            tx_fee: 0,
+            key_encoding: TransactionPublicKeyEncoding::Compressed,
+            signature: MessageSignature([0u8; 65]),
+        };
+
+        let tx = StacksTransaction::new(
+            TransactionVersion::Testnet,
+            TransactionAuth::Standard(TransactionSpendingCondition::Singlesig(sig)),
+            TransactionPayload::new_smart_contract("test", "test", Some(ClarityVersion::Clarity3))
+                .unwrap(),
+        );
+        event_dispatcher.process_new_mempool_txs(vec![tx]);
+
+        _m.assert();
+    }
+
+    #[test]
+    fn test_event_dispatcher_nakamoto_block_filtering() {
+        let dir = tempdir().unwrap();
+        let working_dir = dir.path().to_path_buf();
+
+        // Create a mock server
+        let mut server = mockito::Server::new();
+        let _m = server
+            .mock("POST", "/new_mempool_tx")
+            .with_status(200)
+            .expect_at_most(1)
+            .create();
+
+        let mut event_dispatcher = EventDispatcher::new();
+        let config = EventObserverConfig {
+            endpoint: server.url().strip_prefix("http://").unwrap().to_string(),
+            events_keys: vec![EventKeyType::BlockProposal],
+            timeout_ms: 1000,
+        };
+        event_dispatcher.register_observer(&config, working_dir);
+
+        let nakamoto_block = NakamotoBlock {
+            header: NakamotoBlockHeader::empty(),
+            txs: vec![],
+        };
+        event_dispatcher.process_mined_nakamoto_block_event(
+            0,
+            &nakamoto_block,
+            0,
+            &ExecutionCost::max_value(),
+            vec![],
+        );
+
+        _m.assert();
+    }
+
+    #[test]
+    fn test_event_dispatcher_nakamoto_block_filtering_empty() {
+        let dir = tempdir().unwrap();
+        let working_dir = dir.path().to_path_buf();
+
+        // Create a mock server
+        let mut server = mockito::Server::new();
+        let _m = server
+            .mock("POST", "/new_mempool_tx")
```

Comment:
As I mentioned above - I think we need to _always_ send `/new_block` proposals, so this test might be moot. Regardless, the path here should be `/new_block`.

Assuming we _do_ make it so that `/new_block` is always sent, it would be good to modify this test such that we ensure `/new_block` is called event with an empty `event_keys`.